app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,23 @@
+class SessionsController < ApplicationController
+  def new
+    flash[:success] = 'You are already logged in!' if current_user
+
+    redirect_to '/dashboard'
+  end
+
+  def create
+    if user = User.find_by(email: params[:email])
+      if user.authenticate(params[:password])
+        session[:user_id] = user.id
+        flash[:success] = "You have successfully logged in!"
+        redirect_to '/dashboard'
+      else
+        flash[:error] = 'Invalid email or password, please try again.'
+        redirect_to root_path
+      end
+    else
+      flash[:error] = 'Invalid email or password, please try again.'
+      redirect_to root_path
+    end
```

Comment:
This can be refactored a bit using the built-in error messages:

`user = User.find_by(email: params[:email])` 
`    if user && user.authenitcate(params[:password])` 
`        #... ` 
`    else`
`        flash[:error] = user.errors.full_messages.to_sentence`
`    end`